python/torch_mlir/__init__.py

```diff
@@ -465,25 +465,84 @@ def compile(model: torch.nn.Module,
 
     return _lower_mlir_module(verbose, output_type, mb.module)
 
+
 def run_via_iree(module, *model_args):
+    from torch.utils._pytree import tree_map
+    import numpy as np
     try:
-        import iree_torch
-    except:
-        print("ERROR: Failed to import iree_torch")
+        import iree.runtime as ireert
+        import iree.compiler as ireec
+    except Exception as e:
+        print("ERROR: Failed to import iree")
         print("pip install iree-compiler iree-runtime")
-        print("git clone https://github.com/iree-org/iree-torch && pip install iree-torch --no-deps")
+        print(e)
         sys.exit(1)
 
-    backend = LinalgOnTensorsTosaBackend()
     run_pipeline_with_repro_report(
             module,
             f"builtin.module(func.func({TOSA_TO_LINALG_FUNC_PIPELINE}))",
             "Lowering TOSA backend contract to Linalg-on-Tensors backend contract")
 
     print("Loading inference function into IREE")
-    iree_vmfb = iree_torch.compile_to_vmfb(
-        module, "llvm-cpu")
-    invoker = iree_torch.load_vmfb(iree_vmfb, "llvm-cpu")
+    
+    target_backend = "llvm-cpu"
+    cuda_llvm_target_arch = None
+    extra_args = []
+    if cuda_llvm_target_arch is not None:
```

Comment:
Is this really relevant if `cuda_llvm_target_arch` is set above?

Reply:
Right, I will simplify the code.